cfpb/consumerfinance.gov
# Update to DS@0.21.3 to fix numerous issues

The transition classes in the global search and mega menu had the `u-is-animating` CSS class stuck on the DOM, when it should be removed when they are both not animating. This PR pulls in the latest from the design system, which includes these updates that fix this issue:

https://github.com/cfpb/design-system/pull/1556
https://github.com/cfpb/design-system/pull/1564

The changes fix two issues on mobile where the global search was shifting the whole layout when opened and the global overlay was not extending full height of the available area. 

This PR also changes the behavior of clicking the mega menu and global search slightly, whereby a click of either cancels and reverses any expanding/collapsing currently taking place.

This PR also includes using the Summary component from the DS, which injects the data-hook attributes, so those are removed from the markup.

Also, includes fixes to arabic right-to-left layouts:

https://github.com/cfpb/design-system/pull/1574
https://github.com/cfpb/design-system/pull/1582

## Changes

- Replace cfgov FlyoutMenu and behavior with version supplied by the design-system.
- Use the Summary component from the DS. 
- Use `behavior.js` from the DS.
- Fix issue where whole page was shifting when opening the global search on mobile.
- Fix issue where the global overlay on mobile was not covering the visible area and had a gap at the bottom that was uncovered when the mega-menu or global search were open.
- Downcase custom event names to align with native event naming conventions.
- Update header related cypress and unit tests.
- Fixes alignment issue with right-to-left layouts in notifications and info-units.

## How to test this PR

1. Pull branch, `yarn build`, and check that the global search and mega menu animations work as before (can be expanded/collapsed across desktop and mobile).
2. Check an ask page, such as http://localhost:8000/ask-cfpb/what-effect-will-shopping-for-an-auto-loan-have-on-my-credit-en-763/ to ensure the summary component still works on mobile as expected.
3. Check a TDP page, such as http://localhost:8000/consumer-tools/educator-tools/youth-financial-education/teach/activities/ and ensure expandables work without error.

## Screenshots

Before:

<img width="832" alt="Screen Shot 2023-01-20 at 4 32 59 PM" src="https://user-images.githubusercontent.com/704760/213811915-717b8ad3-b90c-47b8-be9f-027c067f1d15.png">

After:

<img width="771" alt="Screen Shot 2023-01-20 at 4 33 48 PM" src="https://user-images.githubusercontent.com/704760/213811927-df7840a2-eafc-4a5e-8f82-e3f7b693b6d7.png">
